protocol/messenger_communities.go

```diff
@@ -841,6 +897,90 @@ func (m *Messenger) RequestToJoinCommunity(request *requests.RequestToJoinCommun
 	return response, nil
 }
 
+func (m *Messenger) EditSharedAddressesForCommunity(request *requests.RequestToJoinCommunity) (*MessengerResponse, error) {
+	logger := m.logger.Named("EditSharedAddressesForCommunity")
+	if err := request.Validate(); err != nil {
+		logger.Debug("request failed to validate", zap.Error(err), zap.Any("request", request))
+		return nil, err
+	}
+
+	community, err := m.communitiesManager.GetByID(request.CommunityID)
+	if err != nil {
+		return nil, err
+	}
+
+	// verify wallet password is there
+	if request.Password == "" {
+		return nil, errors.New("password is necessary to use this function")
+	}
```

Comment:
I tried the inheritance, but turns out Go doesn't do polymorphism (at least not easily), so I just copied the Class and pass the individual properties to the `getAccountsToShare` function instead of the object. It works fine.